core/java/android/app/Activity.java

```diff
@@ -6931,6 +6931,42 @@ public String getLaunchedFromPackage() {
         return ActivityClient.getInstance().getLaunchedFromPackage(getActivityToken());
     }
 
+    /**
+     * Returns if the permission is granted on app that initially launched this activity.
+     *
+     * <p>In order to receive the launching app's package name permission state,
+     * at least one of the following has to be met:
+     * <ul>
+     *     <li>The app must call {@link ActivityOptions#setShareIdentityEnabled(boolean)} with a
+     *     value of {@code true} and launch this activity with the resulting
+     *     {@code ActivityOptions}.
+     *     <li>The launched activity has the same uid as the launching app.
+     *     <li>The launched activity is running in a package that is signed with the same key
+     *     used to sign the platform (typically only system packages such as Settings will
+     *     meet this requirement).
+     *     <li>The current activity is from a system browser.</li>
+     * </ul>.
+     *
+     * These are superset of requirements for {@link #getLaunchedFromUid()}; if any of these are
+     * met, then these methods can be used to obtain permission state of the launching
+     * app. If none are met, then {@code SecurityException} is thrown.
+     *
+     * <p>Note, even if the above conditions are not met, the launching app's identity may
+     * still be available from {@link #getCallingPackage()} if this activity was started with
+     * {@code Activity#startActivityForResult} to allow validation of the result's recipient.
+     *
+     * @return the permission state of a permission if it's granted
+     *
+     * @see ActivityOptions#setShareIdentityEnabled(boolean)
+     * @see #getLaunchedFromUid()
+     * @hide
+     */
+    @UnsupportedAppUsage
+    @Nullable
+    public boolean doesLaunchedFromPackageHavePermission(String permission) {
```

Comment:
`checkLaunchedFromPackagePermission` would be a better name I think, similar to `Context.checkCallingPermission()`